### ISSUE TITLE
[Bug]: Expected Pimcore\Translation\Translator, Symfony\Component\Translation\DataCollectorTranslator given

### ISSUE THREAD

**Issue by Blackbam** [pimcore/pimcore#9980]
### Expected behavior

Happens after "composer update" when opening the default page. There should not be an exception, it happens in pure Pimcore code (cache cleared).

### Actual behavior

```
TypeError:
Pimcore\Bundle\CoreBundle\EventListener\TranslationDebugListener::__construct(): Argument #1 ($translator) must be of type Pimcore\Translation\Translator, Symfony\Component\Translation\DataCollectorTranslator given, called in /var/www/var/cache/dev/ContainerPcPWxTH/App_KernelDevDebugContainer.php on line 2917

  at /var/www/vendor/pimcore/pimcore/bundles/CoreBundle/EventListener/TranslationDebugListener.php:31
  at Pimcore\Bundle\CoreBundle\EventListener\TranslationDebugListener->__construct(object(DataCollectorTranslator), 'pimcore_debug_translations')
     (/var/www/var/cache/dev/ContainerPcPWxTH/App_KernelDevDebugContainer.php:2917)
  at ContainerPcPWxTH\App_KernelDevDebugContainer->getTranslationDebugListenerService()
     (/var/www/var/cache/dev/ContainerPcPWxTH/App_KernelDevDebugContainer.php:1710)
  at ContainerPcPWxTH\App_KernelDevDebugContainer->ContainerPcPWxTH\{closure}()
     (/var/www/vendor/symfony/event-dispatcher/EventDispatcher.php:245)
  at Symfony\Component\EventDispatcher\EventDispatcher->sortListeners('kernel.request')
     (/var/www/vendor/symfony/event-dispatcher/EventDispatcher.php:76)
  at Symfony\Component\EventDispatcher\EventDispatcher->getListeners('kernel.request')
     (/var/www/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:290)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->preProcess('kernel.request')
     (/var/www/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:145)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (/var/www/vendor/symfony/http-kernel/HttpKernel.php:133)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (/var/www/vendor/symfony/http-kernel/HttpKernel.php:79)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (/var/www/vendor/symfony/http-kernel/Kernel.php:199)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (/var/www/public/index.php:35)                
     
```

### Steps to reproduce

It seems to happen after `composer update` in the Pimcore core with version 10.1. Maybe some issue with dependency injection? Please have a look at the stack trace.

**Comment by dvesh3** [comment]
@Blackbam sorry, not able to reproduce with upgrade scenario from 10.0.9 to 10.1.0
![image](https://user-images.githubusercontent.com/5137917/128987142-93534e76-35e6-4db2-a1a7-02fe3d862051.png)

![image](https://user-images.githubusercontent.com/5137917/128987641-16da4a6d-1e0f-4764-a6b7-2633589e0e3b.png)

**Comment by shaselboeck** [comment]
The [error](https://github.com/pimcore/pimcore/discussions/9950) also occurs with my installation, but only in debug mode. I upgraded from 10.0.9 to 10.1.

**Comment by brusch** [comment]
@shaselboeck @Blackbam can maybe someone who experiences the issue have a look into it? Thanks a lot!

**Comment by Blackbam** [comment]
> @shaselboeck @Blackbam can maybe someone who experiences the issue have a look into it? Thanks a lot!

Will soon try to do the update again and try to provide more details.

**Comment by Humb3l** [comment]
> > @shaselboeck @Blackbam can maybe someone who experiences the issue have a look into it? Thanks a lot!
> 
> 
> 
> Will soon try to do the update again and try to provide more details.

Are there any new findings on this topic?

**Comment by Blackbam** [comment]
@Humb3l Well I have solved it by overriding the class TranslationDebugListener with an empty class doing nothing:

This is my services.yaml:

```
    #
    # Temporary Compatibility Fixes
    #
    Pimcore\Bundle\CoreBundle\EventListener\TranslationDebugListener:
        class: App\PimcoreCompatibility\TranslationDebugListener
```

This is of course not a solution to the problem, it is a hack which eliminates the debug listener essentially.

**Comment by Humb3l** [comment]
@Blackbam 

Ah okay, I have actually tried that as well. However, I run into an 'Symfony/Contracts/Translation/TranslatorInterface Service or alias has been removed ' error after logging in.

Have you stumbled across the same error?

**Comment by Blackbam** [comment]
@Humb3l Nope - nothing like this.

My overriden class looks like this:

```
<?php

declare(strict_types=1);
namespace App\PimcoreCompatibility;

use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\HttpKernel\Event\RequestEvent;
use Symfony\Component\HttpKernel\KernelEvents;

/**
 * @internal
 */
class TranslationDebugListener implements EventSubscriberInterface
{

    /**
     * {@inheritdoc}
     */
    public static function getSubscribedEvents()
    {
        return [
            KernelEvents::REQUEST => 'onKernelRequest',
        ];
    }

    public function onKernelRequest(RequestEvent $event)
    {
       # fix
    }
}
```

**Comment by nexus23** [comment]
I had the same problem
pimcore 10.0.8 to 10.2.1
the routine of _composer update_
adds some Symfony\Bundle in
config/bundles.php
why?
i deleted them and cache:clear
et voilá

**Comment by dvesh3** [comment]
@Blackbam @Humb3l  @nexus23 @shaselboeck could any of you confirm if this PR https://github.com/pimcore/pimcore/pull/10612 fixes the issue? thanks!

**Comment by Blackbam** [comment]
@dvesh3 Unfortunatly not for me, still need the fix I described above.

@nexus23 Can you please specify which bundle you think is responsible for the issue? Or did you remove all bundles from the Symfony\Bundle namespace? For me it does not seem to change anything.

**Comment by dvesh3** [comment]
@Blackbam I would just revert the constructor property promotion changes then to fix it https://github.com/pimcore/pimcore/pull/9912/files#diff-72bf4c4040dea3191f620fd46fb82b6f7332e017c8084dfe6f8c56412e58e279

could please check reverting it fixes the issue? thanks!

**Comment by Blackbam** [comment]
@dvesh3 Very unlikely to make a different as those are only syntactical? I furthermore did them in my local - not dockerized - Pimcore development installation, the problem we are talking about here has been arising since v10.1.0 after downloading dependencies to the customers projects.

As my fix works it must be related to that translations class and it only seems to hapen in dev mode.

**Comment by jdreesen** [comment]
I don't think this should be closed yet, as #10612 doesn't seem to be the real fix.

**Comment by brusch** [comment]
@jdreesen thanks for the hint - this was closed automatically by GitHub because  #10612 was merged 😊

**Comment by fashxp** [comment]
Dear community, it's been a while since that issue was updated the last time, so we decided to close it for now. If you're still interested in contributing, feel free to add comments here or contact us before creating a pull request to discuss next steps.

Thanks for your understanding.
